TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ftc16072/Mechanisms/MecannumDrive.java

```diff
@@ -27,27 +30,39 @@ public void init(HardwareMap HwMap){
         FrontRightMotor = HwMap.get(DcMotor.class, "front_right_motor");
         FrontLeftMotor = HwMap.get(DcMotor.class, "front_left_motor");
 
-        BackLeftMotor.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
-        BackRightMotor.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
-        FrontRightMotor.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
-        FrontLeftMotor.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
 
 
         FrontLeftMotor.setDirection(DcMotorSimple.Direction.REVERSE);
         BackLeftMotor.setDirection(DcMotorSimple.Direction.REVERSE);
 
     }
-    public void convert_to_field(double forward, double right, double rotate){
```

Comment:
I would strongly recommend that the gyro be in its own mechanism and the field relative is either part of the robot (where it can access the imu and the mecanumDrive mechanisms) or is in a util class.

Splitting things up so each class only does things related to that class allows for much easier debugging and is a hallmark of a professional.

Reply:
fixed